src/rest/entries.ts

```diff
@@ -0,0 +1,72 @@
+import { EntryProps } from 'contentful-management/types';
+import { ContentFields } from 'contentful-management/types';
+
+import { SysProps, Entity, ContentType } from '../types';
+import { isPrimitiveField, updatePrimitiveField } from '../utils';
+
+const mergeFields = (
+  fields: ContentFields[],
+  dataFromPreviewApp: any,
+  updateFromEntryEditor: any,
+  locale: string
+): Entity & { sys: SysProps } => {
+  const mergedObj = { ...dataFromPreviewApp };
+
+  for (const field of fields) {
+    const name = field.apiName ?? field.name;
+
+    if (isPrimitiveField(field) || field.type === 'RichText') {
+      updatePrimitiveField(dataFromPreviewApp.fields, updateFromEntryEditor, name, locale);
+      mergedObj.fields[name] = dataFromPreviewApp.fields[name];
+    } else if (field.type === 'Link') {
+      // update single ref field
+    } else if (field.type === 'Array' && field.items?.type === 'Link') {
+      // update multi ref field
+    }
+  }
+
+  return mergedObj;
+};
+
+function updateNestedRef(
+  fields: ContentFields[],
+  dataFromPreviewApp: Entity & { sys: SysProps },
+  updateFromEntryEditor: EntryProps,
+  locale: string
+): Entity & { sys: SysProps } {
+  if (dataFromPreviewApp.fields) {
+    const updatedFields = { ...dataFromPreviewApp.fields };
+
+    for (const [fieldName, field] of Object.entries(dataFromPreviewApp.fields)) {
+      if (field.sys && field.sys.id === updateFromEntryEditor.sys.id) {
+        updatedFields[fieldName] = mergeFields(fields, field, updateFromEntryEditor, locale);
+        return { ...dataFromPreviewApp, fields: updatedFields };
+      } else if (typeof field === 'object') {
```

Comment:
Is there any other way to know whether we should call this `updateNestedRef`? just a little confused about what is going on here.

Reply:
It's to update the fields inside of references, if they have been resolved. It's recursive and will try to find the id of the reference and compare it to the id of the update that is coming from the webapp and then updating it

Reply:
I'm probably going to change this a little bit anyways, as I kind of need it also for the asset updates - sorry!